clippy_lints/src/missing_const_for_fn.rs

```diff
@@ -0,0 +1,120 @@
+use rustc::hir;
+use rustc::hir::intravisit::FnKind;
+use rustc::hir::{Body, Constness, FnDecl};
+// use rustc::mir::*;
+use crate::utils::{is_entrypoint_fn, span_lint};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use rustc_mir::transform::qualify_min_const_fn::is_min_const_fn;
+use syntax::ast::NodeId;
+use syntax_pos::Span;
+
+/// **What it does:**
+///
+/// Suggests the use of `const` in functions and methods where possible.
+///
+/// **Why is this bad?**
+///
+/// Not using `const` is a missed optimization. Instead of having the function execute at runtime,
```

Comment:
That's not quite true. This optimization will still happen for non-const fn. The major use case is crates forgetting to use `const fn` will unnecessarily limit their consumers in what they can do. This is similar to forgetting to implement `PartialEq` or similar things on types where it might be helpful to have that.